### PR TITLE
fix(rust): Non-compliant Parquet list element name

### LINKED ISSUE #17100: Parquet file writer uses non-compliant list element field name
### Checks

- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the [latest version](https://crates.io/crates/polars) of Polars.

### Reproducible example

```python
import polars as pl
import pyarrow.parquet as pq

df = pl.DataFrame(
    {
        "a": [[1, 2], [1, 1, 1]],
    }
)

df.write_parquet("example.parquet", use_pyarrow=False)
print("with polars")
print(pq.read_schema("example.parquet"))
print()
df.write_parquet("example.parquet", use_pyarrow=True)
print("with pyarrow")
print(pq.read_schema("example.parquet"))
```
```
with polars
a: large_list<item: int64>
  child 0, item: int64

with pyarrow
a: large_list<element: int64>
  child 0, element: int64
```

### Log output

_No response_

### Issue description

When generating Parquet files using Polars with `use_pyarrow=False` (using the polars parquet writer), the list element field name is set to `item` instead of `element`. This appears to be non-compliant with the Parquet specification for nested types.

According to the [Parquet specification](https://github.com/apache/parquet-format/blob/master/LogicalTypes.md#lists), the correct field name for the single item in a LIST should be `element`.

This can cause issues when working with other libraries or tools that expect Parquet files to follow the specification. For example, [when trying to add these files to an Apache Iceberg table using pyiceberg](https://github.com/apache/iceberg-python/issues/716), it results in errors due to the unexpected field name.

### Expected behavior

The issue arises because when writing out Parquet files, the schema data types are converted to arrow format here:

https://github.com/pola-rs/polars/blob/main/crates/polars-core/src/datatypes/dtype.rs#L575

However, perhaps the confusion arises because in arrow, the List single element name is often `item` not `element`.

https://arrow.apache.org/docs/format/Columnar.html#recordbatch-message

```python
import pyarrow as pa

py_list = pa.array([[1, 2, 3], [1, 2]])
print(py_list.type)
```
```
list<item: int64>
```

I do not think we want to change the default for all arrow conversions, perhaps another flag similar to `pl_flavor` for `is_parquet` and then accordingly set the `item` strings to `element` so the resulting parquet file matches the spec. 

### Installed versions

<details>

```
--------Version info---------
Polars:               0.20.15
Index type:           UInt32
Platform:             Linux-5.10.218-186.862.amzn2int.x86_64-x86_64-with-glibc2.39
Python:               3.12.3 (main, Apr  9 2024, 08:09:14) [GCC 13.2.0]

----Optional dependencies----
adbc_driver_manager:  <not installed>
cloudpickle:          <not installed>
connectorx:           <not installed>
deltalake:            <not installed>
fastexcel:            <not installed>
fsspec:               <not installed>
gevent:               <not installed>
hvplot:               <not installed>
matplotlib:           3.8.4
numpy:                1.26.4
openpyxl:             <not installed>
pandas:               2.2.1
pyarrow:              16.0.0
pydantic:             <not installed>
pyiceberg:            <not installed>
pyxlsb:               <not installed>
sqlalchemy:           <not installed>
xlsx2csv:             <not installed>
xlsxwriter:           <not installed>
```
</details>

### PR DESCRIPTION
**Pull request by coastalwhite**
Fixes #17100.

### CODE REVIEW

File: crates/polars-parquet/src/arrow/write/schema.rs [hunk 1]
```diff
@@ -390,21 +390,26 @@ pub fn to_parquet_type(field: &Field) -> PolarsResult<ParquetType> {
...
-            vec![ParquetType::from_group(
-                "list".to_string(),
-                Repetition::Repeated,
-                None,
-                None,
-                vec![to_parquet_type(f)?],
+        | ArrowDataType::LargeList(f) => {
+            let mut f = f.clone();
+            f.name = "element".to_string();
+
```
Comment (coastalwhite) [inline crates/polars-parquet/src/arrow/write/schema.rs:396]: This is the actual change. The rest is all pass-by changes.

**Comment by ritchie46** [comment]
Can you add a python test?

File: py-polars/tests/unit/io/test_parquet.py [hunk 2]
```diff
@@ -1244,3 +1244,21 @@ def test_parquet_record_batches_pyarrow_fixed_size_list_16614(tmp_path: Path) ->
...
+    (
+        pl.DataFrame(
+            {
+                "a": [[1, 2], [1, 1, 1]],
+            },
+            schema={"a": pl.List(pl.Int64)},
+        ).write_parquet(filename, use_pyarrow=False)
+    )
+
+    schema_str = pq.read_schema(filename)
```
Comment (nameexhaustion) [inline py-polars/tests/unit/io/test_parquet.py:1262]: check `type(pq.read_schema(filename))` 😄

Comment (coastalwhite) [reply py-polars/tests/unit/io/test_parquet.py:1262]: :man_facepalming: I am stupid lol

**Pull request merged**